internal/controller/compute/statefulserverset/serverset_controller.go

```diff
@@ -148,3 +166,32 @@ func extractSSetFromSSSet(sSSet *v1alpha1.StatefulServerSet) *v1alpha1.ServerSet
 func getSSetName(cr *v1alpha1.StatefulServerSet) string {
 	return fmt.Sprintf("%s-%s", cr.Name, cr.Spec.ForProvider.Template.Metadata.Name)
 }
+
+// Delete - deletes the server k8s client and waits until it is deleted
+func (k *kubeServerSetController) Delete(ctx context.Context, name, namespace string) error {
+	condemnedServer, err := k.Get(ctx, name, namespace)
+	if err != nil {
+		return err
+	}
+	if err := k.kube.Delete(ctx, condemnedServer); err != nil {
+		return fmt.Errorf("error deleting server %w", err)
+	}
+	return kube.WaitForResource(ctx, kube.ResourceReadyTimeout, k.isServerDeleted, condemnedServer.Name, namespace)
+}
+
+func (k *kubeServerSetController) isServerDeleted(ctx context.Context, name, namespace string) (bool, error) {
+	k.log.Info("Checking if Serverset is deleted", "name", name, "namespace", namespace)
+	obj := &v1alpha1.Server{}
```

Comment:
&v1alpha1.ServerSet{}

Reply:
this actually explains a bug I had while testing. thanks!